File: python/src/a_mcp/in_process.py
'''In-process transport: the tool calls the host directly.

This is the only asset superseded when a later MCP-backed transport is swapped in, and it
survives as a fast test double. Everything above the transport interface is reused.
'''

from a_mcp.capability import AuditCapability
from a_mcp.host import AuditHost
from a_mcp.transport import AttemptResponse


class InProcessTransport:
    '''Direct in-process transport implementing the AuditTransport protocol.'''

    def __init__(self, host: AuditHost) -> None:
        '''Bind the transport to a host audit subsystem.'''
        self._host = host
        # end def

    def negotiate(self) -> AuditCapability:
        '''Return the host-declared audit capability.'''
        return self._host.negotiate()
        # end def

    def send_attempt(self, event: dict) -> AttemptResponse:
        '''Forward audit/attempt to the host.'''
        return self._host.handle_attempt(event)
        # end def

    def send_outcome(self, event: dict) -> None:
        '''Forward audit/outcome to the host.'''
        self._host.handle_outcome(event)
        # end def
    # end class

File: reference/typescript/src/transport/inProcess.ts
import type { AuditEvent } from '../schema/event.js';
import type { AuditCapability, NegotiationResult } from '../schema/capability.js';
import type { AuditHost } from '../host/auditHost.js';
import type { AttemptResponse, AuditTransport } from './transport.js';

// In-process transport: the tool calls the host directly. Swapped for a binding (§6.4, §6.5) in a
// real deployment; survives as a fast test double.
export class InProcessTransport implements AuditTransport {
  constructor(private readonly host: AuditHost) {}

  negotiate(offered: AuditCapability): NegotiationResult {
    return this.host.negotiate(offered);
  }

  async sendAttempt(event: AuditEvent): Promise<AttemptResponse> {
    return this.host.handleAttempt(event);
  }

  async sendOutcome(event: AuditEvent): Promise<void> {
    this.host.handleOutcome(event);
  }
}
